piezo_web_app/PiezoWebApp/src/services/spark_job/spark_job_service.py

```diff
@@ -181,6 +182,37 @@ def submit_job(self, body):
                 'message': message
             }
 
+    def tidy_jobs(self):
+        api_response = self.get_jobs(label=None)
+        if api_response['status'] != StatusCodes.Okay.value:
+            return api_response
+        dict_of_jobs = api_response['spark_jobs']
+        jobs_untouched = {}
+        jobs_tidied = {}
+        jobs_failed_to_process = {}
+
+        for job in dict_of_jobs:
```

Comment:
If there are a lot of jobs then this could be very slow (and runs the risk that jobs may be deleted by a separate call before this method gets to them). Please take a look at running asynchronous co-routines https://docs.python.org/3/library/asyncio-task.html

Reply:
Implemented but required downgrading tornado to 4.5.3 as similar issue to: https://github.com/nteract/hydrogen/issues/1357

Reply:
Note downgrading tornado has required changing all cases of HttpClientError to HttpError